software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -20,49 +20,73 @@
 
 import java.util.Collection;
 
+import brooklyn.enricher.Enrichers;
 import brooklyn.entity.Entity;
 import brooklyn.entity.basic.AbstractEntity;
 import brooklyn.entity.basic.Entities;
 import brooklyn.entity.group.DynamicCluster;
 import brooklyn.entity.proxying.EntitySpec;
 import brooklyn.entity.trait.Startable;
+import brooklyn.event.AttributeSensor;
+import brooklyn.event.basic.Sensors;
 import brooklyn.location.Location;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
 public class RedisClusterImpl extends AbstractEntity implements RedisCluster {
 
-    protected RedisStore master;
-    protected DynamicCluster slaves;
+    private AttributeSensor<RedisStore> MASTER = Sensors.newSensor(RedisStore.class, "redis.master");
```

Comment:
In this case the attributes are used exclusively internally in lieu of storing master and slaves as non-persistent member variables, so I have left them as `private static`.  However, if you feel that they should be `public` anyway, I will make that change.